### PR TITLE
[ELASTIC-136] Add module options/settings for elasticsearch

### PR DESCRIPTION
**Pull request by mbdcos**

### CODE REVIEW

File: frameworks/elastic/universe/config.json [hunk 1]
```diff
@@ -326,6 +299,143 @@
...
+          "type": "string",
+          "default": "elastic"
+        },
+        "health_user_password": {
+          "description": "Password to use with health_user for X-Pack authentication, if enabled. Note that you are not setting the password here. You are telling the framework the credentials to use when sending in health check requests.",
+          "type": "string",
+          "default": "changeme"
+        },
+        "gateway_recover_after_time": {
+          "description": "f the expected number of nodes is not achieved, the recovery process waits for the configured amount of time before trying to recover regardless. Defaults to 5m if one of the expected_nodes settings is configured.",
```
Comment (loren) [inline frameworks/elastic/universe/config.json]: If

File: frameworks/elastic/universe/config.json [hunk 2]
```diff
@@ -326,6 +299,143 @@
...
+          "type": "boolean",
+          "default": true
+        },
+        "network_tcp_keep_alive": {
+          "description": "Enable or disable TCP keep alive. Defaults to true.",
+          "type": "boolean",
+          "default": true
+        },
+        "network_tcp_reuse_address": {
+          "description": "Should an address be reused or not. Defaults to true on non-windows machines.",
```
Comment (loren) [inline frameworks/elastic/universe/config.json]: windows? ;)

File: frameworks/elastic/universe/config.json [hunk 3]
```diff
@@ -326,6 +299,143 @@
...
+          "type": "integer",
+          "default": 3
+        },
+          "http_cors_enabled": {
+          "description": "Enable or disable cross-origin resource sharing, i.e. whether a browser on another origin can execute requests against Elasticsearch. Set to true to enable Elasticsearch to process pre-flight CORS requests.",
+          "type": "boolean",
+          "default": true
+        },
+          "http_cors_max_age": {
+          "description": "Browsers send a \"preflight\" OPTIONS-request to determine CORS settings. max-age defines how long the result should be cached for. Defaults to 1728000 (20 days)",
```
Comment (loren) [inline frameworks/elastic/universe/config.json]: s/max-age/http_cors_max_age

File: frameworks/elastic/universe/config.json [hunk 4]
```diff
@@ -326,6 +299,143 @@
...
+          "type": "integer",
+          "default": 1728000
+        },
+        "http_cors_allow_credentials": {
+          "description": "Whether the Access-Control-Allow-Credentials header should be returned. Note: This header is only returned, when the setting is set to true. Defaults to false.",
```
Comment (loren) [inline frameworks/elastic/universe/config.json]: No comma in "This header is only returned, when"

File: frameworks/elastic/universe/config.json [hunk 5]
```diff
@@ -326,6 +299,143 @@
...
+          "type": "boolean",
+          "default": false
+        },
+        "http_detailed_errors_enabled": {
+          "description": "nables or disables the output of detailed error messages and stack traces in response output. Note: When set to false and the error_trace request parameter is specified, an error will be returned; when error_trace is not specified, a simple message will be returned. Defaults to true.",
```
Comment (loren) [inline frameworks/elastic/universe/config.json]: E

File: frameworks/elastic/universe/config.json (hunk 4 above)
Comment (mbdcos) [reply frameworks/elastic/universe/config.json]: https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-http.html

**Comment by mbdcos** [comment]
Thanks a lot @loren

**Comment by mbdcos** [comment]
All elastic index tests are failing. Http module changes cause this mess. Not sure which option is responsible. Disabling all (will be investigated in another PR) for now.

**Pull request merged**